src/asm/main.c

```diff
@@ -268,10 +268,23 @@ static struct option const longopts[] = {
 
 static void print_usage(void)
 {
-	printf(
-"usage: rgbasm [-EhLVvw] [-b chars] [-Dname[=value]] [-g chars] [-i path]\n"
-"              [-M dependfile] [-o outfile] [-p pad_value]\n"
-"              [-r recursion_depth] [-W warning] [-w] file.asm\n");
+	fputs(
+"usage: rgbasm [<options>] <file>...\n"
+"    -b, --binary-digits <digits>   change the chars used for binary constants\n"
+"    -D, --define <name>[=<value>]  add a string define (`EQUS')\n"
+"    -E, --export-all               export all labels\n"
+"    -g, --gfx-chars <chars>        change the chars used for gfx constants\n"
+"    -h, --halt-without-nop         do not automatically add `nop' after `halt'\n"
+"    -i, --include <path>           add the directory to the include search list\n"
+"    -L, --preserve-ld              prevent auto-optimizing `ld' to `ldh'\n"
+"    -M, --dependfile <path>        set the output dependency file\n"
+"    -o, --output <path>            set the output object file\n"
+"    -p, --pad-value <value>        set the value to use for `ds'\n"
+"    -r, --recursion-depth <depth>  set max recursion depth before aborting\n"
+"    -V, --version                  print RGBASM version and exit\n"
+"    -v, --verbose                  report more information\n"
+"    -W, --warning                  enable or disable warnings\n"
+"    -w                             disable all warnings\n", stderr);
```

Comment:
What used to be a three‐line gist now takes up two thirds of a 24‐line terminal, and will only get longer as options are added. Inevitably this becomes a monster like `less --help` or `rsync --help`.

My opinion regarding the intersection of usage statements and documentation is that usage statements are a quick reference in case someone forgets the order of options or arguments, and one’s instinct when more detail is needed should be to open the clear, concise, and complete manpage necessarily included with the program.

Reply:
I have picked a few commands and ran their `--help`: `ls`, `gcc` and `python` printed such monsters, `git commit` just pulled up its manual, and `ps` had a very short notice that also ended with the following prompt: `For more details see ps(1).`.

Considering the `gcc` help only listed a couple options, maybe only the more popular ones could be listed (`-E`, maybe `-h`, `-i`, `-o`, `-p`, and `-W`), plus the man prompt at the end?

Reply:
BSD practice is for usage to match the synopsis of the manpage, and list every option and argument, but not what they do (because that’s what `DESCRIPTION` is for). This is what the programs print on my machine:

    usage: ls [-1AaCcdFfgHhikLlmnopqRrSsTtux] [file ...]

    usage: ps [-AaceHhjkLlmrSTuvwx] [-M core] [-N system] [-O fmt] [-o fmt] [-p pid]
              [-t tty] [-U username] [-W swap]

Reply:
Okay then.

Reply:
That said, how would that go with long options? Don't mention them at all?

Reply:
Do computers with 24-line terminals without scrollback capability even support testing the output of RGBLINK in an emulator? Even so, if we substantially restrict the information provided by `--help` for the benefit of those still using such terminals, we might need to document solutions to these:

- How to install `man` on Windows
- How to install manpages on both Linux, FreeBSD, and macOS for a user who has built the program and installed it to the user's home directory (as opposed to the superuser installing it system-wide or obtaining it from the distribution's repository)

Might it be worthwhile to do different things for `-h` and `--help`? The former could emit the existing terse format and the latter a more detailed description with one line per option.

Reply:
See more(1).

Reply:
`-h` is already taken, anyways. (`--halt-without-nop`)

As for the man pages, there are the HTML versions, though they're only for releases (see #458).

Reply:
> computers with 24-line terminals without scrollback capability

It’s less about the technical capabilities of the terminal and more about a terminal page being a decent analog for human comprehensibility. I don’t know about you, but I find reaching for the mouse and scrolling back to the top, then visually scanning for options in a gigantic list (no search!), to rarely teach me what I want to know.

> See more(1).

Ah yes, leading to one of my (least) favorite sequences when dealing with a new Linux program:

    $ rsync
    [spits out 155 lines of disorganized output, simultaneously too much to handle
    and too little to tell me how the program actually works]
    $ rsync | less
    [oops, it printed to stderr!]
    $ rsync 2>&1 | less
    [there we go]

Give me a decent manpage any day.

Reply:
> Considering the `gcc` help only listed a couple options, maybe only the more popular ones could be listed (`-E`, maybe `-h`, `-i`, `-o`, `-p`, and `-W`), plus the man prompt at the end?

I think I’d prefer the existing usage plus a man prompt. But if you have a strong preference for this version, I won’t oppose it.

Reply:
If the problem is a "reasonable size" one, then I believe it can boil down to long options: should they be listed in the `--help` as well? If so, the synopsis' size would probably be enough to warrant nothing else (but a man prompt), but otherwise I would probably aim to document a handful of options while staying under 10 lines. Does that sound reasonable?

Reply:
That sounds reasonable.

Reply:
I made choices in 6959b76, hopefully they are popular. I tried to pick the six (five for RGBFIX due to the two-line description) most useful options.
This yielded 10 lines of help, plus 2 at the end for the man/online help (as suggested by @pinobatch above) prompt. I don't think removing two more options from the list to get to 10 lines would make sense, so I broke the barrier.

Reply:
Going to assume this is fine, merging.